### DISCUSSION TITLE
system logs

### DISCUSSION THREAD

**Discussion by rye2020** [microsoft/WSL#6493] [General] [answered]
Where does WSL2/init maintain logs for system messages?  systemd maintains logs in /var/log.  Where are WSL2 logs?

**Comment by benhillis** [comment, marked as answer]
WSL2 logs error directly to /dev/kmsg.

**Reply by rye2020** [reply]
thank you, @benhillis

**Comment by dylan-chong** [comment]
Hi, can you access the system logs from windows? I cannot access `/dev/kmsg` when WSL is frozen
